bindings/js/test.html

```diff
@@ -0,0 +1,90 @@
+
+<!doctype html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>KaRaMeL main driver</title>
+    <script type="application/javascript">
+      var my_modules = ["WasmSupport", "FStar", "LowStar_Endianness", "LowStar", "Hacl_Impl_Blake2_Constants", "Hacl_Hash_Base", "Hacl_Hash_Blake2", "Hacl_Hash_MD5", "Hacl_Hash_SHA1", "Hacl_Hash_SHA2", "EverCrypt_TargetConfig", "EverCrypt_EverCrypt", "Vale", "Hacl_Streaming", "EverCrypt_Hash", "Hacl_SHA3", "Hacl_Chacha20", "Hacl_Salsa20", "Hacl_IntTypes_Intrinsics", "Hacl_Bignum", "Hacl_Bignum25519_51", "Hacl_Curve25519_51", "Hacl_Ed25519", "Hacl_Poly1305_32", "Hacl_NaCl", "Hacl_P256", "Hacl_HMAC", "Hacl_HKDF", "Hacl_Chacha20Poly1305_32", "Hacl_HPKE_Curve51_CP32_SHA256", "Hacl_HPKE_Curve51_CP32_SHA512", "Steel_Reference", "Hacl_GenericField32", "Hacl_Bignum256", "Hacl_Bignum4096", "Hacl_Chacha20_Vec32", "Hacl_Bignum4096_32", "Hacl_GenericField64", "Hacl_Bignum32", "Hacl_Bignum256_32", "Hacl_Bignum64", "Hacl_EC_Ed25519"];
+      var my_print = msg =>
+        document.getElementById("terminal").appendChild(
+          document.createTextNode(msg + "\n"));
+    </script>
+    <script type="application/javascript" src="browser.js"></script>
+    <script type="application/javascript" src="loader.js"></script>
+    <script type="application/javascript" src="api.js"></script>
+    <script type="application/javascript">
+      window.addEventListener("load", () => {
+        // We demonstrate how to write a sample program that runs directly
+        // within the browser, without node or webpack. Run `http-server .`,
+        // then navigate to `test.html`. See `test2.js` for a more comprehensive
+        // API demonstration.
+
+        // Test helpers
+        // ------------
+
+        function buf2hex(buffer) {
+          return Array.prototype.map.call(new Uint8Array(buffer), function(x) {
+            return ('00' + x.toString(16)).slice(-2);
+          }).join('');
+        }
```

Comment:
I think using spread syntax (`[...buffer].map(function (x) ...)`) would be more readable here, instead of `Array.prototype.map.call(...)`.

Also, why not using fat arrows & template strings? For instance:
```js
const buf2hex = buffer => [...buffer].map(x => `00${x.toString(16)}`.slice(-2)).join('')
```

Reply:
@W95Psp I think you are just pointing out my advanced age and the fact that I have not caught up with the latest JS syntax innovations. Thanks for dragging me into the future ;-)

Reply:
I took into account your other comments but this one doesn't work because ArrayBuffer is not iterable and so spread syntax does not work for it. I'm marking this as resolved, but please feel free to suggest something else. Thanks.

Reply:
Ah, yeah, there is this thing of views over buffers, I did not read carefuly enough the original code. I think ``const buf2hex = buffer => [...new Uint8Array(buffer)].map(x => `00${x.toString(16)}`.slice(-2)).join('')`` would work then!

Reply:
That worked, thanks!